docs/src/test/scala/docs/http/scaladsl/server/PlayRoutesComparisonSpec.scala

```diff
@@ -0,0 +1,137 @@
+      }
+      // #fixed-test
+    }
+
+    "show reading a long" in {
+      val clientById: Route =
+        // #long
+        (get & pathPrefix("client" / LongNumber)) { id =>
+          complete(Clients.get(id))
+        }
+      // #long
+
+      // #long-test
+      Get("/client/321433") ~> clientById ~> check {
+        responseAs[String] shouldEqual "clientB"
+      }
+      // #long-test
+    }
+
+    "use multiple segments" in {
+      def download(name: String) = s"$name: file contents"
+
+      val files: Route =
+        // #segments
+        (get & pathPrefix("files" / Segments)) { names =>
+          val name = names.mkString("/")
+          complete(download(name))
+        }
+      // #segments
+
+      // #segments-test
+      Get("/files/images/logo.png") ~> files ~> check {
+        responseAs[String] shouldEqual "images/logo.png: file contents"
+      }
+      // #segments-test
+    }
+
+    "require a parameter" in {
+      def getPage(name: String) = s"The requested [$name]."
+
+      val pageParameter: Route =
+        // #mandatory-parameter
+        (get & parameter("page")) { page =>
+          complete(getPage(page))
+        }
+      // #mandatory-parameter
+
+      // #mandatory-parameter-test
+      Get("/?page=example.txt") ~> pageParameter ~> check {
+        responseAs[String] shouldEqual "The requested [example.txt]."
+      }
+      // #mandatory-parameter-test
+    }
+
+    "accept a parameter" in {
+      def listAll(version: Option[String]) = version.fold("ff")(_ => "aa,bb,cc")
+
+      val optionalPageParameter: Route =
+        // #optional-parameter
+        get {
+          path("api" / "list-all")(parameter("version".?) { version =>
+            complete(listAll(version))
+          })
+        }
+      // #optional-parameter
+
+      // #optional-parameter-test
+      Get("/api/list-all?version=3.0") ~> optionalPageParameter ~> check {
+        responseAs[String] shouldEqual "aa,bb,cc"
+      }
+      Get("/api/list-all") ~> optionalPageParameter ~> check {
+        responseAs[String] shouldEqual "ff"
+      }
+      // #optional-parameter-test
+    }
+
+    "accept a parameter list" in {
+      def listItems(items: Iterable[String]) = items.mkString(",")
+
+      val itemParameterList: Route =
+        // #parameter-list
+        // TODO:
+        // Note that implicit conversions are not applicable because they are ambiguous:
+        //  both method augmentString in object Predef of type (x: String)scala.collection.immutable.StringOps
+        //  and method _string2NR in trait ToNameReceptacleEnhancements of type (string: String)akka.http.scaladsl.common.NameReceptacle[String]
+        //  are possible conversion functions from String("item") to ?{def *: ?}
+        //         path("api" / "list-items")(parameters("item".*) { items =>
+        get {
+          path("api" / "list-items")(parameters(ToNameReceptacleEnhancements._string2NR("item").*) { items =>
```

Comment:
If we want to keep PR's independent, we could keep showing the symbol syntax here for now until we introduce syntax for this that works with strings?

Reply:
Did that now. Other PR incoming...